### PR TITLE
Features/ #629 gas stores and generators agg

### LINKED ISSUE #629: Aggregate generators and stores for CH4
In some cases, the CH4- and H2-voronoi contain more than one CH4 generator and/or more than one CH4 store (non extendable). These could be aggregated.

### PR DESCRIPTION
**Pull request by AmeliaNadal**
Fixes #629 . 

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted. 
- [x] the branch was merged into the `continuous-integration/run-everything-over-the-weekend`- branch.
- [x] the workflow is running successful in `test mode`.
- [x] the workflow is running successful in `Everything` mode.

### CODE REVIEW

**Comment by CarlosEpia** [comment]
Hi @AmeliaNadal, hi @fwitte. I ran egon-data in the morning and found that in the table grid.egon_etrago_generator there are some generators without values for the columns "bus" and "carrier". Could you check if the mentioned generators are created in this branch?

**Comment by CarlosEpia** [comment]
Thanks Amélia!

File: src/egon/data/datasets/gas_aggregation.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    """
+    # Connect to local database
+    engine = db.engine()
+
+    components = [
+        # Generator
+        {
+            "columns": "scn_name, generator_id, bus, p_nom, carrier",
+            "table": "egon_etrago_generator",
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:41]: You should import "egon.data.datasets" to make reference to tables in the egon-data db. In this way, if it is necessary to change the name of a table, we just need to change that file.

File: src/egon/data/datasets/gas_aggregation.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    ]
+
+    for comp in components:
+        df = db.select_dataframe(
+            f"""SELECT {comp["columns"]}
+                    FROM grid.{comp["table"]}
+                    WHERE scn_name = '{scn_name}' 
+                    AND carrier = 'CH4';"""
+        )
+        print(df)
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:71]: Maybe we can delete this

File: src/egon/data/datasets/gas_aggregation.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        df = df.groupby(["bus", "carrier"]).agg(comp["strategies"])
+        print(df)
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:73]: Maybe we can delete this

File: src/egon/data/datasets/gas_aggregation.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        # Clean table
+        db.execute_sql(
+            f"""DELETE FROM grid.{comp["table"]}
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:77]: You should import and use "egon.data.datasets" and call the tables from there.

File: src/egon/data/datasets/gas_aggregation.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        # Clean table
+        db.execute_sql(
+            f"""DELETE FROM grid.{comp["table"]} 
+                        WHERE carrier = 'CH4'
+                        AND scn_name = '{scn_name}';"""
+        )
+
+        # Insert data to db
+        df.to_sql(
+            comp["table"],
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:84]: You should import and use "egon.data.datasets" and call the tables from there.

File: src/egon/data/datasets/gas_aggregation.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            engine,
+            schema="grid",
```
Comment (CarlosEpia) [inline src/egon/data/datasets/gas_aggregation.py:86]: You should import and use "egon.data.datasets" and call the tables from there.

**Review by CarlosEpia — COMMENTED** [review]
It looks like it is working properly. Once you check the comments and run it again, this branch can be merged.

File: src/egon/data/datasets/gas_aggregation.py (hunk 3 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/f06ff0ca55a0140a9bd6f8737f481697cd5052b2.

File: src/egon/data/datasets/gas_aggregation.py (hunk 4 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/f06ff0ca55a0140a9bd6f8737f481697cd5052b2.

File: src/egon/data/datasets/gas_aggregation.py (hunk 1 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/f06ff0ca55a0140a9bd6f8737f481697cd5052b2.

File: src/egon/data/datasets/gas_aggregation.py (hunk 5 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/f06ff0ca55a0140a9bd6f8737f481697cd5052b2.

File: src/egon/data/datasets/gas_aggregation.py (hunk 6 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/f06ff0ca55a0140a9bd6f8737f481697cd5052b2.

File: src/egon/data/datasets/gas_aggregation.py (hunk 3 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/be2a9e0d21ebfbedc22b3903fefdf0a4e1d11336.

File: src/egon/data/datasets/gas_aggregation.py (hunk 2 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_aggregation.py]: Fixed with https://github.com/openego/eGon-data/pull/632/commits/be2a9e0d21ebfbedc22b3903fefdf0a4e1d11336.

**Pull request merged**